library/certs.c

```diff
@@ -345,33 +1197,40 @@ const char mbedtls_test_cas_pem[] =
 const size_t mbedtls_test_cas_pem_len = sizeof( mbedtls_test_cas_pem );
 #endif
 
-/* List of all available CA certificates */
+/* List of all available CA certificates in DER and PEM format */
 const char * mbedtls_test_cas[] = {
 #if defined(TEST_CA_CRT_RSA_SHA1)
     mbedtls_test_ca_crt_rsa_sha1,
+    mbedtls_test_ca_crt_rsa_sha1_der,
```

Comment:
This change, and the three below, lead to a curious failure in `ssl_server2`. It will attempt to parse all the CAs in this array, and won't accept any to fail. However, if the PEM parsing is disable, this is bound to fail.

I think 
a) the DER CAs could be only added conditionally here, but
b) it may be more reasonable to allow `ssl_server2` to fail on some CAs (from the perspective of the code clarity). Or
c) it may be better to have two `test_cas` arrays, one with PEM and one with DER data and only carry on if at least one of them has been parsed completely and successfully. This way, we may determine presence of bad CAs, and allow for error if the PEM parser is not available.

From the reviewer's perspective, I will not insist on any of the proposed solutions. I am just tossing ideas around.

Reply:
I think that's an excellent suggestion. I will use a separate array for DER encoded CRTs instead.